scripts/triggers.js

```diff
@@ -2,22 +2,55 @@ var firstBeat = true;
 var firstSample = true;
 var newLaptop = false;
 
+// A trigger function usually does a check
+// and performs an action if said check is true
+// It will return true if action happened and false otherwise
+// Example
+/*
+function exampleTrigger() {
+  if (some conditions) {
+    trigger actions
+    return true;
+  }
+  return false;
+}
+ */
+
+var triggerFnSet = new Set([firstBeatTrigger, firstSampleTrigger, newLaptopTrigger]);
+
 function checkTriggers() {
-	// TODO: Figure out more scalable way to add trigger events...
-	if (game.player.beats == 1 && firstBeat) {
+  triggerFnSet.forEach(function(triggerFn) {
+    if (triggerFn()) {
+      triggerFnSet.delete(triggerFn);
+    }
+  });
+}
+
+function firstBeatTrigger() {
+  if (game.player.beats == 1 && firstBeat) {
```

Comment:
I don't think we need the firstBeat var anymore. If this function is in the triggerSet, then this has to be our first beat

Reply:
Same applies to the firstSample var below